base/applications/mspaint/atlimagedx.h

```diff
@@ -17,6 +17,8 @@ class CImageDx : public CImage
         BitmapSetResolution = NULL;
     }
 
+    using CImage::FileTypeFromExtension;
```

Comment:
Why is this needed?

Reply:
That is a `protected` member.

Reply:
`friend` keyword wouldn't work here?

Reply:
Doing that will expose a design flaw.

Reply:
```c
#ifdef _MSPAINT_H
    friend ULONG CMainWindow::OpenMailerWithAttachmentW(LPCWSTR pszPathName);
#endif
```
This code won't work because `CMainWindow` is not declared.

Reply:
```c
#ifdef _MSPAINT_H
    public:
#else
    protected:
#endif
```
This will work.

Reply:
`using CImage::FileTypeFromExtension;`

This is simple and the best.

Reply:
> ```c
> #ifdef _MSPAINT_H
>     public:
> #else
>     protected:
> #endif
> ```
> 
> This will work.

But this is inelegant. atlimage.h should stay MS-compatible, not have hacks for internal applications.

Reply:
https://jira.reactos.org/browse/CORE-19008